base/unicode/utf8.jl

```diff
@@ -221,29 +221,26 @@ function reverse(s::UTF8String)
             pos += 1
         end
     end
-    UTF8String(buf)
+    String(buf)
 end
 
 ## outputting UTF-8 strings ##
 
-write(io::IO, s::UTF8String) = write(io, s.data)
+write(io::IO, s::String) = write(io, s.data)
 
 ## transcoding to UTF-8 ##
 
-utf8(x) = convert(UTF8String, x)
-convert(::Type{UTF8String}, s::UTF8String) = s
-convert(::Type{UTF8String}, s::ASCIIString) = UTF8String(s.data)
-convert(::Type{SubString{UTF8String}}, s::SubString{ASCIIString}) =
-    SubString(utf8(s.string), s.offset+1, s.endof+s.offset)
+utf8(x) = convert(String, x)
+convert(::Type{String}, s::String) = s
```

Comment:
So removing this actually causes a bootstrapping problem – I haven't gotten to the bottom of why.